Assets/Scripts/TrapdoorController.cs

```diff
@@ -0,0 +1,83 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TrapdoorController : MonoBehaviour
+{
+    #region Serialized Public Fields
+    [Header("Collapse Time")] 
+    [SerializeField] public float collapsePlatTimer;
+    [SerializeField] public float restorePlatTimer;
+    [SerializeField] public bool isLeftWall;
+    [SerializeField] public float motorForce;
+    [SerializeField] public float motorSpeed;
+    #endregion
+
+    #region Private Properties
+    private HingeJoint2D _hinge;
+    private JointMotor2D _motor;
+    private float _deltaCollapseTimer;
+    private float _deltaRestoreTimer;
+    private bool _isOnPlatform;
+    #endregion
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        _deltaCollapseTimer = collapsePlatTimer;
+        _deltaRestoreTimer = restorePlatTimer;
+        _hinge = GetComponent<HingeJoint2D>();
+        _motor = _hinge.motor;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(_isOnPlatform) 
+        {
+            _deltaCollapseTimer-=Time.deltaTime;
+
+            if(_deltaCollapseTimer<=0) 
+            {
+                //instantly disable then re-enable current platform's motor to set variables at the same time
+                _hinge.useMotor = false;
+                _motor.motorSpeed = motorSpeed;
+                _motor.maxMotorTorque = motorForce;
+                _hinge.motor = _motor;
+                _hinge.useMotor = true;
```

Comment:
You don't actually need to set/unset `_hinge.useMotor`: the reason why the motor isn't updating is because `JointMotor2D` is a struct and thus a value type, and thus be reassigned (similarly with `Vector2`/`Vector3`s).